### PR TITLE
[java] Multi-LoRA support

### PR DESCRIPTION
**Pull request by Craigacp**
### Description
Java parts of Multi-LoRA support - #22046.

### Motivation and Context
API equivalence with Python & C#.

### CODE REVIEW

File: java/src/main/java/ai/onnxruntime/OrtLoraAdapter.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  }
+
+  /**
+   * Creates an instance of OrtLoraAdapter.
+   *
+   * @param absoluteAdapterPath path to the adapter file that is going to be memory mapped.
+   * @throws OrtException If the native call failed.
+   * @return An OrtLoraAdapter instance.
+   */
+  public static OrtLoraAdapter create(String absoluteAdapterPath) throws OrtException {
```
Comment (yuslepukhin) [inline java/src/main/java/ai/onnxruntime/OrtLoraAdapter.java:105]: >absoluteAdapterPath [](http://example.com/codeflow?start=45&length=19)

I had to relax this requirement and allow relativer paths in other APIs.

File: java/src/main/java/ai/onnxruntime/OrtLoraAdapter.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    return create(absoluteAdapterPath, null);
+  }
+
+  /**
+   * Creates an instance of OrtLoraAdapter.
+   *
+   * @param absoluteAdapterPath path to the adapter file that is going to be memory mapped.
```
Comment (yuslepukhin) [inline java/src/main/java/ai/onnxruntime/OrtLoraAdapter.java:112]: >absoluteAdapterPath [](http://example.com/codeflow?start=12&length=19)

Same here

**Review by yuslepukhin — CHANGES_REQUESTED** [review]
:clock1:

File: java/src/main/java/ai/onnxruntime/OrtLoraAdapter.java (hunk 1 above)
Comment (Craigacp) [reply java/src/main/java/ai/onnxruntime/OrtLoraAdapter.java]: Fixed.

File: java/src/main/java/ai/onnxruntime/OrtLoraAdapter.java (hunk 2 above)
Comment (Craigacp) [reply java/src/main/java/ai/onnxruntime/OrtLoraAdapter.java]: Fixed.

**Review by yuslepukhin — DISMISSED** [review]
:shipit:

**Comment by Craigacp** [comment]
Some of the failures were when it hit spotless as I forgot to run it after changing the variable name. There was a failure in Windows & Mac CI pipelines which I had to download the logs to see and that seemed to be in a flatbuffers header, so I'm not sure how that's related to this change.

**Review by yuslepukhin — APPROVED** [review]
:shipit:

**Pull request merged**